File: frontend/src/app/enums/application-status.ts
export enum ApplicationStatus {
  SENT = 'SENT',
  INTERVIEW = 'INTERVIEW',
  REJECTED = 'REJECTED',
  ACCEPTED = 'ACCEPTED',
}

export const APPLICATION_STATUS_LABELS: { [key in ApplicationStatus]: string } = {
  [ApplicationStatus.SENT]: 'Sent',
  [ApplicationStatus.INTERVIEW]: 'Interview',
  [ApplicationStatus.REJECTED]: 'Rejected',
  [ApplicationStatus.ACCEPTED]: 'Accepted',
};

export const APPLICATION_STATUS_COLORS: { [key in ApplicationStatus]: string } = {
  [ApplicationStatus.SENT]: 'primary',
  [ApplicationStatus.INTERVIEW]: 'accent',
  [ApplicationStatus.REJECTED]: 'warn',
  [ApplicationStatus.ACCEPTED]: 'success',
};

File: backend/src/main/java/com/tracker/job_application_tracker/enums/ApplicationStatus.java
package com.tracker.job_application_tracker.enums;

/**
 * Enum representing the different statuses of a job application
 * 
 * SENT = Application sent
 * INTERVIEW = Interview scheduled/obtained
 * REJECTED = Application rejected
 * ACCEPTED = Offer accepted
 */
public enum ApplicationStatus {
    SENT("Sent"),
    INTERVIEW("Interview"),
    REJECTED("Rejected"),
    ACCEPTED("Accepted");
    
    private final String displayName;
    
    // Enum constructor
    ApplicationStatus(String displayName) {
        this.displayName = displayName;
    }
    
    // Getter for display name
    public String getDisplayName() {
        return displayName;
    }
}